src/ustar_header_parser.e

```diff
@@ -19,169 +20,212 @@ inherit
 feature -- Parsing
 
 	parse_block (block: MANAGED_POINTER; pos: INTEGER)
-			-- Parse ustar-header in `block' starting at position `pos'
+			-- Parse ustar-header in `block' starting at position `pos'.
 		local
-			current_field: STRING_8
+			l_field: detachable STRING_8
+			l_header: like last_parsed_header
+			err: BOOLEAN
 		do
-			create current_header.make
-
-			-- parse filename
-			-- FIXME: Implement filename splitting
-			current_field := parse_string (block, pos + {TAR_CONST}.tar_header_name_offset, {TAR_CONST}.tar_header_name_length)
-			if (current_field.is_empty) then
-				current_header := Void
-			end
-
-			if (attached current_header as header) then
-				header.set_filename (create {PATH}.make_from_string (current_field))
+			create l_header.make
+
+				-- parse "filename"
+				-- FIXME: Implement filename splitting
+			if not err then
```

Comment:
replaced current_... by l_... locals
and keep l_header attached, but manage error with "err" local variable.

Then only at the end l_header is assigned to current_header (or renamed last_parsed_header)